aws/resource_aws_sns_topic_policy.go

```diff
@@ -92,6 +96,14 @@ func resourceAwsSnsTopicPolicyRead(d *schema.ResourceData, meta interface{}) err
 
 	d.Set("policy", policy)
 
+	arn, ok := attrmap["TopicArn"]
+	if !ok {
+		log.Printf("[WARN] SNS Topic (%q) topic arn not found in attributes", d.Id())
+		d.SetId("")
```

Comment:
Instead of removing the resource from the Terraform state due to a potentially different API response (possible with tooling such as SAM CLI or LocalStack, we should just allow the attribute value to be conditionally set, e.g.

```go
d.Set("arn", attrmap["TopicArn"])
```